cocos/2d/renderer/mesh-buffer.ts

```diff
@@ -152,6 +155,9 @@ export class MeshBuffer {
         this._nextFreeIAHandle = 0;
 
         this._dirty = false;
+
+        this._initIDataCount = 256 * 6;
+        this._initVDataCount = 256 * this._vertexFormatBytes;
```

Comment:
If we want to keep the same `vData` and `iData` buffer, we shouldn't reset them like this.

Reply:
Dear! if you don't reset this._initIDataCount and this._initVData, any time you call 
`public request (vertexCount = 4, indicesCount = 6)`
...
The code will always double this._initVDataCount so it can lead to an infinity value => max length create buffer BUG.
`if (byteOffset > byteLength || indicesOffset > indicesLength) {
            while (byteLength < byteOffset || indicesLength < indicesOffset) {
                this._initVDataCount *= 2;
                this._initIDataCount *= 2;

                byteLength = this._initVDataCount * 4;
                indicesLength = this._initIDataCount;
            }

            this._reallocBuffer();
        }`

Reply:
As the `byteOffset` is being reset to 0, the request would only double the buffer size while the `byteLength` and indices length is consumed out again, as `byteLength` equals to `_initVDataCount  * 4` and `indicesLength` equals to `_initIDataCount`, it's totally reasonable. Just as the implication I mentioned before.

Reply:
@pandamicro you still want to remove the code?
this._initIDataCount = 256 * 6;
this._initVDataCount = 256 * this._vertexFormatBytes;